Cleanup and refactoring

genu opened the pull request:
@rxb3rth This is an ambitious refactoring of the module, while still trying to stay within the original scope.

### Summary
1. Re-organized the `src` folder to align more with the nuxt runtime structure
2. Renamed `useClientFlags()` to `useFeatureFlags()` Since there isn't now a distinction between `client` flags and `server` flags.
3. Using flags on the server is now done through `getFeatureFlags(event)`. This is inspired by `nuxt-auth-utils`. I like this pattern because it's explicit that on the server feature flags are influenced by the context (Internally, the client ultimately calls `getFeatureFlags(event)`)
4. Added an alias so that generated flags can imported using just 
`import type { FlagsSchema } from '#feature-flags'`

### Examples
On the client, we'd do:

```ts
const { isEnabled, flags } = useFeatureFlags()
```

On the server, we can do: 

```ts
export default defineEventHandler(async (event) => {
const { isEnabled } =  await getFeatureFlags(event)

if(isEnabled('experimentalFeature')) {
   ....
}
})

```

Let me know your thoughts on this. I'm sure there is room for improvement.

genu commented on src/module.ts:
```diff
@@ -33,33 +38,21 @@ export default defineNuxtModule<FeatureFlagsConfig>({
-      // @ts-ignore
-      nuxt.options.runtimeConfig.public.featureFlags, options)
-
-    nuxt.options.nitro = nuxt.options.nitro || {}
-    nuxt.options.nitro.imports = nuxt.options.nitro.imports || {}
-    nuxt.options.nitro.imports.presets = nuxt.options.nitro.imports.presets || []
```
I'm not exactly sure why this was needed

genu commented on src/module.ts:
```diff
@@ -14,15 +20,14 @@ export default defineNuxtModule<FeatureFlagsConfig>({
+    // Load feature flags configuration from file so that we can generated types from them
     if (options.config) {
       try {
-        consolador.info('Loading feature flags from:', options.config)
         const { config: configFlags, configFile } = await loadConfigFile(options.config, nuxt.options.rootDir)
-        consolador.info('Loaded feature flags:', configFlags)
         options.flags = defu(options.flags, configFlags || {})
```
I removed the info logs here because this section now is primarily helpful for generating types rather than loading the feature flags for runtime usage.

The actual loading for runtime usage of the flags happens in the server util `feature-flags.ts`

rxb3rth commented:
Everything else according to my tests I see it quite correct and if it is ready I do merge. @genu

genu commented:
LGTM 🚀 

Will need to update docs too

rxb3rth commented:
> LGTM 🚀
> 
> Will need to update docs too

I've added a couple of changes, so don't forget to update your branch.

genu commented:
> I've added a couple of changes, so don't forget to update your branch

I like the `defineFeatureFlagsConfig` helper. I wonder if can autoimport it, or expose it automatically through `#feature-flags` alias

rxb3rth commented:
> > I've added a couple of changes, so don't forget to update your branch
> 
> I like the `defineFeatureFlagsConfig` helper. I wonder if can autoimport it, or expose it automatically through `#feature-flags` alias

I like the idea of autoimport

genu commented:
I'm running into an issue with `c12` and `jiti` used in `config-loader.ts` and in `utils/feature-flags.ts`.

I don't think we need to use `jiti` in `feature-flags.ts` and load the config file again. Instead we should be able to generate the file when the module initializes. Similar how we generate the types.

I'll have another look at it in a few hours.

rxb3rth commented:
> I'm running into an issue with `c12` and `jiti` used in `config-loader.ts` and in `utils/feature-flags.ts`.
> 
> I don't think we need to use `jiti` in `feature-flags.ts` and load the config file again. Instead we should be able to generate the file when the module initializes. Similar how we generate the types.
> 
> I'll have another look at it in a few hours.

In early implementations I had done it as you say, loading it when the module was started, however due to serialization problems I omitted the first step. If you can do it, go ahead.

genu commented:
@rxb3rth I've done a few changes that It seems to now work.

Notable updates:
1. Moved the config loader directly in `module.ts` since now its only done one time.
2. Introduced a `feature-flags.config.ts` in the module that serves as a fallback config in cases where flags are hardcoded in `nuxt.config.ts`
3. When a config file is specified, the default `feature-flags.config.ts` is ignored.

In my testing, it seems to work as expected.

Looking forward to your feedback and testing.

rxb3rth commented:
@genu I have tested it and it works very well, need to update the documentation and it's ready, let me know to make the merge.

genu commented:
🚀 Feel free to merge.
